scale-tests/run_spark_scale_test.sh

```diff
@@ -0,0 +1,226 @@
+    CLUSTER_USERNAME)                       CLUSTER_USERNAM=${VALUE} ;;
+    CLUSTER_PASSWORD)                       CLUSTER_PASSWORD=${VALUE} ;;
+    CLUSTER_URL)                            CLUSTER_URL=${VALUE} ;;
+    KAFKA_CONFIG)                           KAFKA_CONFIG=${VALUE} ;;
+    KAFKA_ZK_CONFIG)                        KAFAKA_ZK_CONFIG=${VALUE} ;;
+    NUM_DISPATCHERS)                        NUM_DISPATCHERS=${VALUE} ;;
+    NUM_CASSANDRA_CLUSTERS)                 NUM_CASSANDRA_CLUSTERS=${VALUE} ;;
+    NUM_FINITE_DISPATCHERS)                 NUM_FINITE_DISPATCHERS=${VALUE} ;;
+    NUM_KAFKA_CLUSTERS)                     NUM_KAFKA_CLUSTERS=${VALUE} ;;
+    NUM_GPU_DISPATCHERS)                    NUM_GPU_DISPATCHERS=${VALUE} ;;
+    NUM_PRODUCER_WORDS)                     NUM_PRODUCER_WORDS=${VALUE} ;;
+    NUM_PRODUCER_WORDS_PER_SECOND)          NUM_PRODUCER_WORDS_PER_SECOND=${VALUE} ;;
+    NUM_PRODUCERS_PER_FINITE_KAFKA)         NUM_PRODUCERS_PER_KAFKA=${VALUE} ;;
+    NUM_CONSUMERS_PER_FINITE_PRODUCER)      NUM_CONSUMERS_PER_PRODUCER=${VALUE} ;;
+    NUM_PRODUCERS_PER_INFINITE_KAFKA)       NUM_PRODUCERS_PER_KAFKA=${VALUE} ;;
+    NUM_CONSUMERS_PER_INFINITE_PRODUCER)    NUM_CONSUMERS_PER_PRODUCER=${VALUE} ;;
+    QUOTA_DRIVERS_CPUS)                     QUOTA_DRIVERS_CPUS=${VALUE} ;;
+    QUOTA_DRIVERS_MEM)                      QUOTA_DRIVERS_MEM=${VALUE} ;;
+    QUOTA_EXECUTORS_CPUS)                   QUOTA_EXECUTORS_CPUS=${VALUE} ;;
+    QUOTA_EXECUTORS_MEM)                    QUOTA_EXECUTORS_MEM=${VALUE} ;;
+    QUOTA_EXECUTORS_GPUS)                   QUOTA_EXECUTORS_GPUS=${VALUE} ;;
+    TEST_NAME)                              TEST_NAME=${VALUE} ;;
+    TEST_S3_BUCKET)                         TEST_S3_BUCKET=${VALUE} ;;
+    TEST_S3_FOLDER)                         TEST_S3_FOLDER=${VALUE} ;;
+    SECURITY)                               SECURITY=${VALUE} ;;
+
+  esac
+
+done
+
+if [ -z $SECURITY ]
+then
+  echo "Need to set SECURITY either to 'strict' or 'permissive'"
+  exit 1
+fi
+
+
+## Get a stable shell 
+git clone git@github.com:mesosphere/spark-build.git
+cd spark-build
+docker build -t mesosphere/dcos-commons:${LOGNAME}-spark scale-tests/
+
+
+## Script to be executed inside the shell
+###### cat > /tmp/spark_scale_script.sh << EOF
+## Setup cluster
+
+dcos cluster setup --insecure --username=${CLUSTER_USERNAME} --password=${CLUSTER_PASSWORD} ${CLUSTER_URL}
+eval "$(ssh-agent)"
+ssh-add -k /ssh/key
+
+## Add Spark stub
+dcos package repo add --index=0 spark-aws https://universe-converter.mesosphere.com/transform?url=https://infinity-artifacts.s3.amazonaws.com/permanent/spark/assets/scale-testing/stub-universe-spark.json
+
+## Deploy dispatchers
+./scale-tests/deploy-dispatchers.py \
+  --quota-drivers-cpus $QUOTA_DRIVERS_CPUS \
+  --quota-drivers-mem $QUOTA_DRIVERS_MEM \
+  --quota-executors-cpus $QUOTA_EXECUTORS_CPUS \
+  --quota-executors-mem $QUOTA_EXECUTORS_MEM \
+  $NUM_DISPATCHERS $TEST_NAME ${TEST_NAME}-${NUM_DISPATCHERS}
```

Comment:
One thing to be careful of is this script currently does not wait for the dispatchers to finish deploying (see "wait_for_deployment=False" in deploy-dispatchers.py). You could change this option while installing, or check the dispatchers are ready right before the job submission. With the former, you might want to check if how long this takes since it would be deploying the dispatchers serially.